web/src/routes/ConsentPage.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { ReactElement, useMemo } from 'react'
+import { useTranslation } from 'react-i18next'
+import styled from 'styled-components'
+
+import { ExternalSourcePermissions } from 'api-client'
+
+import Caption from '../components/Caption'
+import ConsentSection from '../components/ConsentSection'
+import GeneralFooter from '../components/GeneralFooter'
+import GeneralHeader from '../components/GeneralHeader'
+import Layout from '../components/Layout'
+import List from '../components/List'
+import buildConfig from '../constants/buildConfig'
+import useCookie from '../hooks/useCookie'
+import { EXTERNAL_SOURCES_COOKIE_NAME } from '../utils/iframes'
+
+const Description = styled.div`
+  margin-bottom: 24px;
+`
+
+type ConsentPageProps = { languageCode: string }
+const ConsentPage = ({ languageCode }: ConsentPageProps): ReactElement => {
+  const { t } = useTranslation('consent')
+  const { value: sources, updateCookie } = useCookie(EXTERNAL_SOURCES_COOKIE_NAME)
+  const externalSourcePermissions: ExternalSourcePermissions = useMemo(
+    () => (sources ? JSON.parse(sources) : {}),
+    [sources],
+  )
+
+  const onPress = (type: string, allowed: boolean) => {
```

Comment:
Maybe we can give this a more telling name like `setNewCookies`, and `permissionGiven` instead of `allowed`. I got confused about why we're setting all cookies to allowed in the next line 😅

Reply:
i renamed to `updateSourcePermission`and `permissionGiven`